File: src/npc.rs
use inventory_item::InventoryItem;

// a non-player character
#[derive(Debug, Eq, PartialEq)]
pub struct NPC {
    pub name: String,
    pub inventory: Vec<InventoryItem>,
    pub dialogue: String,
}

impl NPC {
    pub fn new(name: String, inventory: Vec<InventoryItem>, dialogue: String) -> NPC {
        NPC {
            name: name,
            inventory: inventory,
            dialogue: dialogue,
        }
    }

    fn inventory(&self) -> &[InventoryItem] {
        &self.inventory[..]
    }

    pub fn inventory_mut(&mut self) -> &mut Vec<InventoryItem> {
        &mut self.inventory
    }
}

File: lib/npc.rb
require 'pry'
# NPC = Non Player Character
class NPC
  attr_reader :name, :dialogue
  attr_accessor :inventory

  def initialize(name, inventory, dialogue)
    @name = name
    @inventory = inventory
    @dialogue = dialogue
  end

  def has_item(item)
    @inventory.map(&:name).include?(item)
  end

  def remove_from_inventory(item)
    @inventory.delete_if { |i| item.name == i.name }
    puts "#{item.name} has been removed from #{self.name}'s inventory."
  end
end
